.github/workflows/ci.yml

```diff
@@ -40,15 +40,25 @@ jobs:
     strategy:
       matrix:
         # Test of these containers
-        container: ["ubuntu-dev:20", "alpine-dev:latest"]
+        container: ["ubuntu-dev:20", "alpine-dev:latest", "ubuntu-dev:24"]
         build-type: [Debug, Release]
         compiler: [{ cxx: g++, c: gcc }]
         cxx_flags: ["-Werror"]
+        sanitizers: ["Plain"]
         include:
           - container: "alpine-dev:latest"
             build-type: Debug
             compiler: { cxx: clang++, c: clang }
             cxx_flags: ""
+            sanitizers: "Plain"
+          - container: "ubuntu-dev:24"
+            build-type: Debug
+            compiler: { cxx: clang++, c: clang }
+            cxx_flags: ""
+            sanitizers: "Sanitizers"
+        exclude:
```

Comment:
why do you add it to the matrix.containers list if you exclude it here?

Reply:
Because we would generate a bunch more job. `exclude` is `filter` on the `matrix` level. GH action syntax is missleading here :(

Reply:
You basically "exclude" all the jobs that run on `ubuntu-24` container and are `not sanitizers`. Why? Because otherwise we would have a `release ubuntu-24`, `debug ubuntu-24`, etc. That way you get only one job `sanitizers + ubuntu 24 with clang`

Reply:
include above is the opposite. it acts as a "decorator" overwritting the matrix value

Reply:
From what I understand include adds and exclude removes. here you remove all the ubuntu-dev:24 defined in the matrix above therefor I think that if you remove it from the matrix it will be enough we will just run them from the include

Reply:
I will try